src/server/scripts/Northrend/IcecrownCitadel/boss_the_lich_king.cpp

```diff
@@ -559,6 +559,10 @@ class boss_the_lich_king : public CreatureScript
                     frostmourne->DespawnOrUnsummon();
                 me->GetMap()->SetZoneOverrideLight(AREA_ICECROWN_CITADEL, LIGHT_FOG, 5000);
                 me->GetMap()->SetZoneWeather(AREA_ICECROWN_CITADEL, WEATHER_STATE_FOG, 0.0f);
+
+                if (Is25ManRaid())
+                    if (Player* player = killer->ToPlayer())
```

Comment:
What if a pet kills him? 💃

Reply:
That would be a funny coincidence, to lose the quest kill credit because of the player pet. Could happen, I presume, so I guess you are saying that this check also needs a player pet check to validate the kill?

Reply:
Talking with Sirikfoll, he remember me other situation...
Tirion can kill LK too..
In the end, i think that best solution is:

```cpp
                if (Is25ManRaid())
                    if (Player* player = me->SelectNearestPlayer(100.0f))
                        player->RewardPlayerAndGroupAtEvent(NPC_THE_LICH_KING_QUEST, player);
```

Reply:
what about `map->getplayers` ?
the less searchers the better 👍

Reply:
Yeah, can work too...
But will iterate each player or just get first?